modules/RTC/RTCUtils.js

```diff
@@ -1287,167 +1287,147 @@ class RTCUtils extends Listenable {
      * @returns {*} Promise object that will receive the new JitsiTracks
      */
     obtainAudioAndVideoPermissions(options = {}) {
-        const self = this;
-
         const dsOptions = {
             ...options.desktopSharingExtensionExternalInstallation,
             desktopSharingSources: options.desktopSharingSources
         };
 
-        return new Promise((resolve, reject) => {
-            const successCallback = function(stream) {
-                resolve(handleLocalStream(stream, options.resolution));
-            };
+        options.devices = options.devices || [ 'audio', 'video' ];
+        options.resolution = options.resolution || '720';
 
-            options.devices = options.devices || [ 'audio', 'video' ];
-            options.resolution = options.resolution || '720';
+        const requestingDesktop = options.devices.includes('desktop');
 
-            if (!screenObtainer.isSupported()
-                && options.devices.indexOf('desktop') !== -1) {
-                reject(new Error('Desktop sharing is not supported!'));
-            }
-            if (browser.isFirefox()
-
-                    // XXX The react-native-webrtc implementation that we
-                    // utilize on React Native at the time of this writing does
-                    // not support the MediaStream constructors defined by
-                    // https://www.w3.org/TR/mediacapture-streams/#constructors
-                    // and instead has a single constructor which expects (an
-                    // NSNumber as) a MediaStream ID.
-                    || browser.isReactNative()
-                    || browser.isTemasysPluginUsed()) {
-                const GUM = function(device, s, e) {
-                    this.getUserMediaWithConstraints(device, s, e, options);
-                };
+        if (requestingDesktop && !screenObtainer.isSupported()) {
+            return Promise.reject(
+                new Error('Desktop sharing is not supported!'));
+        }
 
+        let gumPromise;
+
+        // XXX The react-native-webrtc at the time of this writing does not
+        // support MediaStream constructors as defined by
+        // https://www.w3.org/TR/mediacapture-streams/#constructors. Instead it
+        // has a single constructor which expects an NSNumber as a MediaStream
+        // ID. Temasys does not support the constructor at all. So we get audio
+        // and video in two distinct GUM calls and pass them back into a single
+        // object.
+        if (browser.isReactNative() || browser.isTemasysPluginUsed()) {
```

Comment:
can we create a method in "browser" with a meaningful name for this check and put the explanation there ?

Reply:
how about FF ? I may have misread something, but it used to be included in this if. Is it supported now or ?

Reply:
FF should go through new gum flow only now so the FF shouldn't be needed.

Reply:
maybe we could check for FF version and return true only when new flow is supported ? that logic could be in the method mentioned above if you decide to add it.

Reply:
Okay, I tried adding the check. I started getting confused on the boolean flippings so I added an intermediary variable for myself. I'm not sure how useful adding Firefox would be only because all versions of firefox fall under the new gum flow now. But I added it because it's very little work to add it and we can see if it feels good to keep in.